misk-redis/src/test/kotlin/misk/redis/AbstractRedisTest.kt

```diff
@@ -750,4 +758,85 @@ abstract class AbstractRedisTest {
     assertNull(redis.hget("foo", "bar"))
     assertNull(redis["foo"])
   }
+
+  @Test fun zAddTest() {
+    val aMemberBytes = "a".encodeUtf8()
+    val bMemberBytes = "b".encodeUtf8()
+    // add new
+    assertEquals(1L, redis.zadd("bar", 1.0, aMemberBytes));
+    // add new and existing
+    assertEquals(1L, redis.zadd("bar", mapOf(aMemberBytes to 2.0, bMemberBytes to 1.0)))
+    assertEquals(2.0, redis.zscore("bar", "a"))
+    assertEquals(1.0, redis.zscore("bar", "b"))
+
+    // xx: never add new member
+    assertEquals(0L, redis.zadd("foo", 1.0, aMemberBytes, XX))
+
+    redis.zadd("foo", 1.0, aMemberBytes)
+    // nx: never update current member
+    assertEquals(0L, redis.zadd("foo", 2.0, aMemberBytes, NX))
+    assertEquals(1.0, redis.zscore("foo", "a"))
+
+    val scoreMembers = mutableMapOf<ByteString, Double>()
+    scoreMembers[aMemberBytes] = 2.0
+    scoreMembers[bMemberBytes] = 1.0
+    // ch: return count of members not only added, but also updated
+    assertEquals(2L, redis.zadd("foo", scoreMembers, CH))
+
+    // lt: only update existing elements if the new score is less than the current score.
+    redis.zadd("foo", 3.0, aMemberBytes, LT)
+    assertEquals(2.0, redis.zscore("foo", "a"))
+    redis.zadd("foo", 1.0, aMemberBytes, LT)
+    assertEquals(1.0, redis.zscore("foo", "a"))
+    // LT doesn't prevent new members from being added.
+    redis.zadd("foo", 4.0, "c".encodeUtf8(), LT)
+    assertEquals(4.0, redis.zscore("foo", "c"))
+
+    // gt: only update existing elements if the new score is greater than the current score.
+    redis.zadd("foo", 0.0, bMemberBytes, GT)
+    assertEquals(1.0, redis.zscore("foo", "b"))
+    redis.zadd("foo", 2.0, bMemberBytes, GT)
+    assertEquals(2.0, redis.zscore("foo", "b"))
+    // GT doesn't prevent new members from being added.
+    redis.zadd("foo", 4.0, "d".encodeUtf8(), GT)
+    assertEquals(4.0, redis.zscore("foo", "d"))
```

Comment:
You're missing tests for flag combos like `LT XX` or `GT XX`

Reply:
added more tests and organized them better too.